### ISSUE TITLE
Allow AbstractDict for edge attributes

### ISSUE THREAD

**Issue by simonschoelly** [JuliaPlots/GraphRecipes.jl#137]
I noticed that edge attributes can be specified by a `Dict` but not an `AbstractDict`. Would it make sense to allow also `AbstractDict`? Then for example one could use `DefaultDict` from DataStructures.jl to specify some edge attributes.

**Comment by JackDevine** [comment]
I agree, I don't think that I had any really good reasons for choosing `Dict`, other than that is the data structure that I use 99% of the time. Feel free to make a pull request with this feature, otherwise I will add it to my todo list.
